packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -0,0 +1,216 @@
+        });
+      }).toThrowError(/Invalid description length of/);
+    }),
+
+    test('fails portfolio creation with invalid provider name', () => {
+      expect(() => {
+        new Portfolio(stack, 'MyPortfolio', {
+          displayName: 'testPortfolio',
+          providerName: '',
+        });
+      }).toThrowError(/Invalid provider name length/);
+    }),
+
+    test('fails portfolio creation with invalid description length', () => {
+      const description = 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit'.repeat(50);
+
+      expect(() => {
+        new Portfolio(stack, 'MyPortfolio', {
+          displayName: 'testPortfolio',
+          providerName: 'testProvider',
+          description: description,
+        });
+      }).toThrowError(/Invalid description length/);
+    });
+  }),
+
+  describe('portfolio methods and associations', () => {
+    let portfolio: Portfolio;
+
+    beforeEach(() => {
+      portfolio = new Portfolio(stack, 'MyPortfolio', {
+        displayName: 'testPortfolio',
+        providerName: 'testProvider',
+      });
+    });
+
+    test('portfolio with tags', () => {
+      Tags.of(portfolio).add('myTestKey1', 'myTestKeyValue1');
+      Tags.of(portfolio).add('myTestKey2', 'myTestKeyValue2');
+
+      expect(stack).toHaveResourceLike('AWS::ServiceCatalog::Portfolio', {
+        Tags: [
+          {
+            Key: 'myTestKey1',
+            Value: 'myTestKeyValue1',
+          },
+          {
+            Key: 'myTestKey2',
+            Value: 'myTestKeyValue2',
+          },
+        ],
+      });
+    }),
+
+    test('portfolio share', () => {
+      const shareAccountId = '012345678901';
+
+      portfolio.shareWithAccount(shareAccountId);
+
+      expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioShare', {
+        AccountId: shareAccountId,
+      });
+    }),
+
+    test('portfolio share with share tagOptions', () => {
+      const shareAccountId = '012345678901';
+
+      portfolio.shareWithAccount(shareAccountId, {
+        shareTagOptions: true,
+        acceptLanguage: AcceptLanguage.EN,
+      });
+
+      expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioShare', {
+        AccountId: shareAccountId,
+        ShareTagOptions: true,
+        AcceptLanguage: 'en',
+      });
+    }),
+
+    test('portfolio share without share tagOptions', () => {
+      const shareAccountId = '012345678901';
+
+      portfolio.shareWithAccount(shareAccountId, { shareTagOptions: false });
+
+      expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioShare', {
+        AccountId: shareAccountId,
+        ShareTagOptions: false,
+      });
+    }),
+
+    test('portfolio share without explicit share tagOptions', () => {
+      const shareAccountId = '012345678901';
+
+      portfolio.shareWithAccount(shareAccountId);
+
+      expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioShare', {
+        AccountId: shareAccountId,
```

Comment:
If you want to confirm the lack of something in the template, you can use the `ABSENT` constant. [Example](https://github.com/aws/aws-cdk/blob/cbd755298f6ab73e604b96cab7d1e7c5f0dd8af1/packages/%40aws-cdk/cloudformation-include/test/nested-stacks.test.ts#L687-L692).

Reply:
thanks for the tip this will definitely be useful.